src/celeritas/global/ActionLauncher.device.hh

```diff
@@ -62,7 +63,7 @@ launch_action_impl(Range<ThreadId> const thread_range, F execute_thread)
  }
  * \endcode
  */
-template<class F>
+template<class F, int threadsPerBlock = 256, int blocksPerSm = 1>
```

Comment:
Is it possible for these parameters to be obtained from optional `static constexpr int` class data of F? Would that be better by keeping the layout close to the implementation, or worse because it could be convoluted with launcher wrappers?

Also 
```suggestion
template<class F, int T = 256, int B = 1>
```
or `ThreadsPerBlock`.

Reply:
Also I don't think we want BlocksPerSm=1 to be the default; we want no second argument to be the default (maybe need to use variadic template parameters?)

Reply:
I think it makes sense to have a `static constexpr`in F since these values are very much dependent on the kernel. 

For both cuda and hip, the 2nd argument has a default value of 1 so I'd say a default value of 1 doesn't do much because it not applying any constraint, though using variadic templates is probably a better solution.

Reply:
Good point. Default of 1 is good for me.

Reply:
the first argument, `maxThreadsPerBlock` is a constraint though so we might still want to use variadic templates, it also allows to specify the 3rd argument for the CUDA version if needed.